### ISSUE TITLE
Incompatibility between Azure compute instance and AutoML due to xgboost: Check failed: header == serialisation_header_

### ISSUE THREAD

**Issue by marcindulak** [Azure/MachineLearningNotebooks#1421]
I'm running https://github.com/Azure/MachineLearningNotebooks/tree/6f893ff0b40d3d6af94ac43867676a3d9df1fcbd/how-to-use-azureml/automated-machine-learning/regression-explanation-featurization with the "Python 3.6 AzureML" kernel on a default compute instance.

Retrieving the best model fails

```
best_run, fitted_model = remote_run.get_output()
---------------------------------------------------------------------------
XGBoostError                              Traceback (most recent call last)
/anaconda/envs/azureml_py36/lib/python3.6/site-packages/azureml/train/automl/_model_download_utilities.py in _load_automl_model(model_path, suffix)
    176             with open(model_path, "rb") as model_file:
--> 177                 fitted_model = pickle.load(model_file)  # type: Optional[Any]
    178                 return fitted_model

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/xgboost/core.py in __setstate__(self, state)
   1087             _check_call(
-> 1088                 _LIB.XGBoosterUnserializeFromBuffer(handle, ptr, length))
   1089             state['handle'] = handle

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/xgboost/core.py in _check_call(ret)
    188     if ret != 0:
--> 189         raise XGBoostError(py_str(_LIB.XGBGetLastError()))
    190 

XGBoostError: [14:13:18] /mnt/xgboost/src/learner.cc:922: Check failed: header == serialisation_header_: 

  If you are loading a serialized model (like pickle in Python) generated by older
  XGBoost, please export the model by calling `Booster.save_model` from that version
  first, then load it back in current version.  There's a simple script for helping
  the process. See:

    https://xgboost.readthedocs.io/en/latest/tutorials/saving_model.html

  for reference to the script, and more details about differences between saving model and
  serializing.


Stack trace:
  [bt] (0) /anaconda/envs/azureml_py36/lib/libxgboost.so(dmlc::LogMessageFatal::~LogMessageFatal()+0x7c) [0x7eff9669b64c]
  [bt] (1) /anaconda/envs/azureml_py36/lib/libxgboost.so(xgboost::LearnerIO::Load(dmlc::Stream*)+0x66b) [0x7eff967a5f0b]
  [bt] (2) /anaconda/envs/azureml_py36/lib/libxgboost.so(XGBoosterUnserializeFromBuffer+0x5e) [0x7eff9668f7fe]
  [bt] (3) /anaconda/envs/azureml_py36/lib/python3.6/lib-dynload/../../libffi.so.6(+0x69dd) [0x7f0070a379dd]
  [bt] (4) /anaconda/envs/azureml_py36/lib/python3.6/lib-dynload/../../libffi.so.6(+0x6067) [0x7f0070a37067]
  [bt] (5) /anaconda/envs/azureml_py36/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(_ctypes_callproc+0x2ce) [0x7f0070a4fede]
  [bt] (6) /anaconda/envs/azureml_py36/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(+0x13915) [0x7f0070a50915]
  [bt] (7) /anaconda/envs/azureml_py36/bin/python(_PyObject_FastCallDict+0x8b) [0x559e140b6e3b]
  [bt] (8) /anaconda/envs/azureml_py36/bin/python(+0x199c0e) [0x559e1413ec0e]



The above exception was the direct cause of the following exception:

ConfigException                           Traceback (most recent call last)
<ipython-input-17-5d7baab903d5> in <module>
----> 1 best_run, fitted_model = remote_run.get_output()

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/azureml/train/automl/run.py in get_output(self, iteration, metric, return_onnx_model, return_split_onnx_model, **kwargs)
    702                     fitted_model = _download_automl_onnx_model(curr_run, model_name)
    703                 else:
--> 704                     fitted_model = _download_automl_model(curr_run, model_name)
    705 
    706             return curr_run, fitted_model

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/azureml/train/automl/_model_download_utilities.py in _download_automl_model(run, model_name)
     89             # Pass through any exceptions from loading the model.
     90             # Try is used here to ensure we can cleanup the side effect of model downlad.
---> 91             model = _load_automl_model(model_path, suffix)
     92             if model is None:
     93                 # If we can retrieve the automl runtime version, we do so we can inform the user what to install.

/anaconda/envs/azureml_py36/lib/python3.6/site-packages/azureml/train/automl/_model_download_utilities.py in _load_automl_model(model_path, suffix)
    279                 ),
    280                 inner_exception=e
--> 281             ) from e
    282         # If we hit here, we are in new territory - we didn't run into an ImportError
    283         # and we didn't have a known exception type (XGBoostError).

ConfigException: ConfigException:
	Message: The model you attempted to retrieve requires 'xgboost' to be installed at '==0.90'. You have 'xgboost==1.3.3', please reinstall 'xgboost==0.90' (e.g. `pip install xgboost==0.90`) and rerun the previous command.
	InnerException: XGBoostError: [14:13:18] /mnt/xgboost/src/learner.cc:922: Check failed: header == serialisation_header_: 

  If you are loading a serialized model (like pickle in Python) generated by older
  XGBoost, please export the model by calling `Booster.save_model` from that version
  first, then load it back in current version.  There's a simple script for helping
  the process. See:

    https://xgboost.readthedocs.io/en/latest/tutorials/saving_model.html

  for reference to the script, and more details about differences between saving model and
  serializing.


Stack trace:
  [bt] (0) /anaconda/envs/azureml_py36/lib/libxgboost.so(dmlc::LogMessageFatal::~LogMessageFatal()+0x7c) [0x7eff9669b64c]
  [bt] (1) /anaconda/envs/azureml_py36/lib/libxgboost.so(xgboost::LearnerIO::Load(dmlc::Stream*)+0x66b) [0x7eff967a5f0b]
  [bt] (2) /anaconda/envs/azureml_py36/lib/libxgboost.so(XGBoosterUnserializeFromBuffer+0x5e) [0x7eff9668f7fe]
  [bt] (3) /anaconda/envs/azureml_py36/lib/python3.6/lib-dynload/../../libffi.so.6(+0x69dd) [0x7f0070a379dd]
  [bt] (4) /anaconda/envs/azureml_py36/lib/python3.6/lib-dynload/../../libffi.so.6(+0x6067) [0x7f0070a37067]
  [bt] (5) /anaconda/envs/azureml_py36/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(_ctypes_callproc+0x2ce) [0x7f0070a4fede]
  [bt] (6) /anaconda/envs/azureml_py36/lib/python3.6/lib-dynload/_ctypes.cpython-36m-x86_64-linux-gnu.so(+0x13915) [0x7f0070a50915]
  [bt] (7) /anaconda/envs/azureml_py36/bin/python(_PyObject_FastCallDict+0x8b) [0x559e140b6e3b]
  [bt] (8) /anaconda/envs/azureml_py36/bin/python(+0x199c0e) [0x559e1413ec0e]


	ErrorResponse 
{
    "error": {
        "code": "UserError",
        "message": "The model you attempted to retrieve requires 'xgboost' to be installed at '==0.90'. You have 'xgboost==1.3.3', please reinstall 'xgboost==0.90' (e.g. `pip install xgboost==0.90`) and rerun the previous command.",
        "target": "get_output",
        "inner_error": {
            "code": "NotSupported",
            "inner_error": {
                "code": "IncompatibleOrMissingDependency"
            }
        },
        "reference_code": "910310e6-2433-40cd-b597-9ec2950bc1d8"
    }
}
```

It should also be noted that the shell commands within the notebook seem to use the base conda environment, instead of azureml_py36 (see the screenshot below), so it's unclear if attempts to "downgrade" the compute instance to `!pip install --upgrade --force-reinstall xgboost==0.90` without restarting the kernel are feasible.

![Screenshot from 2021-04-07 16-22-12](https://user-images.githubusercontent.com/3178318/113883882-d77ff600-97be-11eb-99b2-556c6984a5a7.png)

Please see similar past issues:

https://github.com/Azure/MachineLearningNotebooks/issues/1124
https://github.com/Azure/MachineLearningNotebooks/issues/1002

and the related feature request https://github.com/Azure/MachineLearningNotebooks/issues/1315. A solution to this recurrent problem may consist of providing a way to version the environments used by both compute instances and the remote AutoML compute targets.

**Comment by marcindulak** [comment]
Additional information: the failure may be limited to the azureml_py36 conda environment.

When starting a terminal from jupyter using New -> Terminal, it seems like azureml_py38 is default environment, and it contains xgboost 0.90. The azureml_py36 conda environment contains xgboost 1.3.3.
See the attached image.

![Screenshot from 2021-04-07 19-55-14](https://user-images.githubusercontent.com/3178318/113912530-85999900-97db-11eb-9acf-9e0cfd39d974.png)

**Comment by CESARDELATORRE** [comment]
@marcindulak This is a known issue with the initial release of Compute Instance for SDK 1.24.0.  
When the issue was found, a new image was released with xgboost==0.90, which is the supported version by AutoML.
  
We just created a new Compute Instance and it does have xgboost==0.90.
You might have tried initially with a Compute Instance created with the original CI image which had that bug/issue.
 
Workarounds are:
1) Manually downgrade xgboost with: pip install xgboost==0.90
or
2) Create a new Compute Instance which should have xgboost==0.90

Closing this issue for now, but re-open if you still are blocked, ok?
